include/boost/process/v2/ext/impl/cwd.ipp

```diff
@@ -139,19 +138,23 @@ filesystem::path cwd(boost::process::v2::pid_type pid, boost::system::error_code
                 filestat *fst = nullptr;
                 STAILQ_FOREACH(fst, head, next) {
                     if (fst->fs_uflags & PS_FST_UFLAG_CDIR) 
-                    {
                         path = filesystem::canonical(fst->fs_path, ec);
-                    }
                 }
                 procstat_freefiles(proc_stat, head);
+                if (errno)
+                    BOOST_PROCESS_V2_ASSIGN_LAST_ERROR(ec)
             }
             else
                 BOOST_PROCESS_V2_ASSIGN_LAST_ERROR(ec)
             procstat_freeprocs(proc_stat, proc_info);
+            if (errno)
+                BOOST_PROCESS_V2_ASSIGN_LAST_ERROR(ec)
         }
         else
             BOOST_PROCESS_V2_ASSIGN_LAST_ERROR(ec)
         procstat_close(proc_stat);
+        if (errno)
```

Comment:
Is this i a guess, or do you know this resets `errno` if it succeeds? Because it might just be a function that never fails.

Reply:
for other people reading this pull request who don't know what i told klemens on slack, these functions never fail and are just wrappers around the free() function, after inspecting FreeBSD's source tree for libprocstat's source code.